File: app/src/main/java/home/oleg/popularmovies/data/network/MovieApi.java
package home.oleg.popularmovies.data.network;

import home.oleg.popularmovies.data.entities.MovieResponse;
import home.oleg.popularmovies.data.entities.ReviewResponse;
import home.oleg.popularmovies.data.entities.VideosResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * Created by Oleg Sheliakin on 06.08.2017.
 * Contact me by email - olegsheliakin@gmail.com
 */

public interface MovieApi {

    @GET(value = "movie/{path}")
    Call<MovieResponse> getMovies(@Path(value = "path") String path);

    @GET(value = "movie/{id}/reviews")
    Call<ReviewResponse> getReviews(@Path(value = "id") Integer id);

    @GET(value = "movie/{id}/videos")
    Call<VideosResponse> getVideos(@Path(value = "id") Integer id);
}

File: app/src/test/java/home/oleg/popularmovies/data/network/MovieApi.kt
package home.oleg.popularmovies.data.network

import DaggerTestMovieAppComponent
import home.oleg.RxSchedulesRule
import home.oleg.popularmovies.data.entities.MovieResponse
import home.oleg.popularmovies.domain.MovieRepository
import okhttp3.mockwebserver.MockWebServer
import org.junit.After
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import retrofit2.Retrofit
import javax.inject.Inject

/**
 * Created by Oleg Sheliakin on 06.08.2017.
 * Contact me by email - olegsheliakin@gmail.com
 */

class MovieApiTest {

    @get:Rule
    val rxSchedulesRule = RxSchedulesRule()

    @Inject
    lateinit var mockWebserver: MockWebServer

    @Inject
    lateinit var movieApi: MovieApi

    @Inject
    lateinit var results: List<MovieResponse.Result>

    @Before
    fun setUp() {
        DaggerTestMovieAppComponent.create().inject(this)
       // mockWebserver.start()
    }

    @Test
    fun shouldReturnPopular() {
        movieApi.getMovies(MovieRepository.Filter.POPULAR.value)
                .map { it.results }
                .test()
                .assertResult(results)
    }

    @After
    fun tearDown() {
        mockWebserver.shutdown()
    }

}
